### ISSUE TITLE
RStudio 1.4.1717 and R 4.1.0 do not respect code chunk options to suppress messages/warnings/errors

### ISSUE THREAD

**Issue by bingh0** [rstudio/rstudio#9436]
### System details

    RStudio Edition : Desktop
    RStudio Version : 1.4.1717
    OS Version      :  macOS 11.3.1
    R Version       :  4.1.0

### Steps to reproduce the problem
1. Create new R Notebook
2. Embed code chunk such as the following

````
```{r}
library(ggplot2)

ggplot(data = pressure, aes(x = temperature, y = pressure)) +
  geom_point() +
  geom_smooth()
```
```{r, echo = FALSE}
library(ggplot2)

ggplot(data = pressure, aes(x = temperature, y = pressure)) +
  geom_point() +
  geom_smooth()
```
```{r, message = FALSE}
library(ggplot2)

ggplot(data = pressure, aes(x = temperature, y = pressure)) +
  geom_point() +
  geom_smooth()
```
```{r, message = FALSE, warning = FALSE}
library(ggplot2)

ggplot(data = pressure, aes(x = temperature, y = pressure)) +
  geom_point() +
  geom_smooth()
```
```{r, message = FALSE, warning = FALSE, error = FALSE}
library(ggplot2)

ggplot(data = pressure, aes(x = temperature, y = pressure)) +
  geom_point() +
  geom_smooth(blah)
```
```{r, echo = TRUE, message = FALSE, warning = FALSE, error = FALSE}
library(ggplot2)

ggplot(data = pressure, aes(x = temperature, y = pressure)) +
  geom_point() +
  geom_smooth()
```
```{r, echo = TRUE, message = FALSE, warning = FALSE, error = FALSE}
library(ggplot2)

ggplot(data = pressure, aes(x = temperature, y = pressure)) +
  geom_point() +
  geom_smooth(blah)
```
````

### Describe the problem in detail
Code chunk options do not appear to be correctly evaluated in html_notebook.

Of note, more than one unexpected behavior can be observed.
1. Messages and errors are not suppressed with code chunk options
2. echo = TRUE is necessary for subsequent code chunks or code will not be shown in rendered document. Code chunk options should be applied for just the specific code chunk.

The behavior is as expected for html_document output (excepting the code chunk with error), when knitted.

### Describe the behavior you expected
Code chunk options should be observed.

- [x] I have read the guide for [submitting good bug reports](https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports).
- [x] I have installed the latest version of RStudio, and confirmed that the issue still persists.
- [] If I am reporting an RStudio crash, I have included a [diagnostics report](https://support.rstudio.com/hc/en-us/articles/200321257-Running-a-Diagnostics-Report).
- [x] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by kevinushey** [comment]
FYI, you can embed R Markdown documents within a code fence using 4 or more balanced backticks; e.g.

```````
`````
```{r}
print("hello, world")
```
`````
```````

**Comment by ronblum** [comment]
@kevinushey Would we expect these chunk options to be observed, given that the code is sent to the console?

**Comment by jgutman** [comment]
I see the regression with the `message` chunk option not being respected starting in 1.4.1717. However, I'm not seeing any other unexpected behavior here. The warning and error chunk options appear to behave as expected with no regressions between versions

**Comment by romainfrancois** [comment]
Verified that the message does not appear either when clicked on the ▶️

![image](https://user-images.githubusercontent.com/2625526/150944736-0bb56c88-ffec-4b93-afcf-83f4711c0428.png)

or when the notebook is preview: 

![image](https://user-images.githubusercontent.com/2625526/150944884-99c5156b-b566-439d-8630-8c28e063305d.png)

**Comment by ronblum** [comment]
Note also that the option `error` doesn't control whether or not error messages are displayed, but whether or not to continue running despite running into errors. For example,

<img width="642" alt="error_handling" src="https://user-images.githubusercontent.com/31009952/151004396-7ea3a5f1-daa3-46bf-81e8-6f8dbca58f92.png">
 
This is described with a little more information at https://yihui.org/knitr/options/#text-output

**Comment by jgutman** [comment]
Yeah, exactly, investigation into this regression revealed that the issue was only with the `message` knitr chunk, which is now addressed in 2022.06.0
